diff --git a/rss2email/config.py b/rss2email/config.py
index 6b5155f..020b66b 100644
--- a/rss2email/config.py
+++ b/rss2email/config.py
@@ -193,6 +193,9 @@ def setup_html2text(self, section='DEFAULT'):
         ('imap-port', str(143)),
         ('imap-ssl', str(False)),       # connect to the IMAP server using SSL
         ('imap-mailbox', 'INBOX'),      # where we should store new messages
+        # Maildir configuration
+        ('maildir-path', '~/Maildir'),
+        ('maildir-mailbox', 'INBOX'),
 
         ### Miscellaneous
         # Verbosity (one of 'error', 'warning', 'info', or 'debug').
diff --git a/rss2email/email.py b/rss2email/email.py
index ef798f7..50e4057 100644
--- a/rss2email/email.py
+++ b/rss2email/email.py
@@ -31,13 +31,15 @@
 from email.mime.text import MIMEText as _MIMEText
 from email.utils import formataddr as _formataddr
 from email.utils import parseaddr as _parseaddr
+import mailbox as _mailbox
 import imaplib as _imaplib
 import io as _io
 import smtplib as _smtplib
 import ssl as _ssl
 import subprocess as _subprocess
 import sys as _sys
 import time as _time
+import os as _os
 
 from . import LOG as _LOG
 from . import config as _config
@@ -199,6 +201,14 @@ def imap_send(message, config=None, section='DEFAULT'):
     finally:
         imap.logout()
 
+def maildir_send(message, config=None, section='DEFAULT'):
+    if config is None:
+        config = _config.CONFIG
+    path = config.get(section, 'maildir-path')
+    mailbox = config.get(section, 'maildir-mailbox')
+    maildir = _mailbox.Maildir(_os.path.join(path, mailbox))
+    maildir.add(message)
+
 def _decode_header(header):
     """Decode RFC-2047-encoded headers to Unicode strings
 
@@ -340,6 +350,8 @@ def send(sender, recipient, message, config=None, section='DEFAULT'):
             config=config, section=section)
     elif protocol == 'imap':
         imap_send(message=message, config=config, section=section)
+    elif protocol == 'maildir':
+        maildir_send(message=message, config=config, section=section)
     else:
         sendmail_send(
             sender=sender, recipient=recipient, message=message,
